hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/SnapshotChainManager.java

```diff
@@ -71,7 +71,27 @@ public SnapshotChainManager(OMMetadataManager metadataManager) {
   private void addSnapshotGlobal(UUID snapshotID,
                                  UUID prevGlobalID) throws IOException {
     // On add snapshot, set previous snapshot entry nextSnapshotID = snapshotID
+    if (globalSnapshotChain.containsKey(snapshotID)) {
+      throw new IllegalStateException(String.format(
+          "Snapshot chain corruption. Snapshot with snapshotId: %s is " +
+              "already present in the the chain.", snapshotID));
+    }
+    if (globalSnapshotChain.size() > 0 && prevGlobalID == null) {
+      throw new IllegalStateException(String.format("Snapshot chain " +
+          "corruption. Adding snapshot %s as head node while there are %d in " +
+          "the global snapshot chain.", snapshotID,
+          globalSnapshotChain.size()));
+    }
     if (prevGlobalID != null && globalSnapshotChain.containsKey(prevGlobalID)) {
```

Comment:
nit: Can we move this `if` block after line 104? Basically move `if` block at line 98-104 before this `if `block

Reply:
done